File: services/29-spring-graphql/src/main/java/ca/yarova/graphql/Application.java
package ca.yarova.graphql;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.graphql.data.method.annotation.QueryMapping;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;
import java.util.Map;

@SpringBootApplication
public class Application {
    public static void main(String[] args) {
        SpringApplication.run(Application.class, args);
    }
}

@Controller
class GraphQlHealthController {
    @QueryMapping
    public String health() { return "ok"; }
}

@RestController
class HttpHealthController {
    @GetMapping("/health")
    public Map<String, String> health() { return Map.of("status", "ok", "version", "1.0.0"); }

    @GetMapping("/health/live")
    public Map<String, String> live() { return Map.of("status", "ok"); }

    @GetMapping("/health/ready")
    public Map<String, String> ready() { return Map.of("status", "ok"); }
}

File: services/18-spring-boot-kotlin/src/main/kotlin/com/example/Application.kt
package com.example

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@SpringBootApplication
class Application

fun main(args: Array<String>) {
    runApplication<Application>(*args)
}

@RestController
class HelloController {
    @GetMapping("/") fun hello() = mapOf("message" to "Hello from Spring Boot Kotlin 3.4", "framework" to "18-spring-boot-kotlin", "version" to "1.0.0")
    @GetMapping("/health") fun health() = mapOf("status" to "ok", "version" to "1.0.0")
    @GetMapping("/health/live") fun liveness() = mapOf("status" to "ok")
    @GetMapping("/health/ready") fun readiness() = mapOf("status" to "ok")
}
